File: src/my_project/easy_problems/from201to250/reverse_only_letters.py
from typing import List 

class Solution:
    def reverseOnlyLetters(self, s: str) -> str:

        n = len(s)

        temp = {i:s[i]  for i in range(len(s)) if not s[i].isalpha()}

        
        s = [s[i] for i in range(len(s)) if s[i].isalpha()]


        s = list(reversed(s))

        for i in range(0,n):
            if i in temp:
                s.insert(i,temp[i])
                
        return ("".join(s))
    
    
solution = Solution()

print(solution.reverseOnlyLetters(s = "ab-cd"))

print(solution.reverseOnlyLetters(s = "a-bC-dEf-ghIj"))

print(solution.reverseOnlyLetters(s = "Test1ng-Leet=code-Q!"))

File: src/my_project_typescript/easy_problems_typescript/from101to150/reverseOnlyLetters.ts
function reverseOnlyLetters(s: string): string {
    let lenS = s.length
    let tempDict: any = {}
    let tempList: Array<string> = []

    for (let i = 0; i < lenS; i++){
        if (s[i].match(/[^a-zA-Z]/)){
            tempDict[i] = s[i]
        }
    }

    for (let i = 0; i < lenS; i++){
        if (s[i].match(/[a-zA-Z]/)){
            tempList.push(s[i])
        }
    }


    tempList = tempList.reverse()

    for (let i = 0; i < lenS; i++){
        if (i.toString() in tempDict){
            tempList = tempList.slice(0,i).concat([tempDict[i]]).concat(tempList.slice(i))
        }
    }
    return tempList.join("")
};

console.log(reverseOnlyLetters("ab-cd"))

console.log(reverseOnlyLetters("a-bC-dEf-ghIj"))

console.log(reverseOnlyLetters("Test1ng-Leet=code-Q!"))
